lib/solidus_affirm/engine.rb

```diff
@@ -8,5 +8,11 @@ class Engine < Rails::Engine
     config.generators do |g|
       g.test_framework :rspec
     end
+
+    config.autoload_paths += %W(#{config.root}/lib)
+
+    initializer "spree.gateway.payment_methods", after: "spree.register.payment_methods" do |app|
```

Comment:
I see no reason for changing this tbh. The sole reason to use this extension is to add the payment method. If users do not need the payment method anymore they remove the extension. If they still need it for legacy reasons, they also need the payment method. This can be disabled in the admin as well.